plugins/modules/network/netscaler/netscaler_nitro_request.py

```diff
@@ -642,10 +642,11 @@ def get_filtered(self):
         if self._module.params['filter'] is None:
             self.fail_module(msg='NITRO filter is undefined.')
 
-        keys = list(self._module.params['filter'].keys())
-        filter_key = keys[0]
-        filter_value = self._module.params['filter'][filter_key]
-        filter_str = '%s:%s' % (filter_key, filter_value)
+        filter_str = ''
+        for key, value in self._module.params['filter'].items():
+            if len(filter_str) != 0:
+                filter_str += '&'
```

Comment:
I'm not sure how the API works, but I would assume that this needs to be percent-encoded. Otherwise the next filter ends up as a different URL parameter.

Reply:
The API makes no sense whatsoever and requires only the parameters themselves to be percent-encoded. The examples I posted in the linked issue are 'redacted' copy-pastes of real requests. It's strange, but it is correct.

Reply:
I guess it actually expects them to be different url parameters, thinking further about it.

Reply:
From the documentation of https://developer-docs.citrix.com/projects/netscaler-nitro-api/en/12.0/configuration/filter/filteraction/#get-all it looks like it must be `','`. Do you have other sources which require `'&'`?

Reply:
Ah no, that's why that API is so confusing. The `get_filtered` operation is not a resource (in the case of your link `filteraction`), but a query, i.e. `/nitro/v1/config/${resource}?filter=key1:value1&key2:value2`. For the example in the issue: https://developer-docs.citrix.com/projects/netscaler-nitro-api/en/12.0/configuration/ssl/sslcertkey/#get-all

Reply:
In that link you pasted, it also uses `,` instead of `&`: `http:// <netscaler-ip-address> /nitro/v1/config/sslcertkey? filter=property-name1:property-val1,property-name2:property-val2 `

Reply:
Ah, you're right. I got confused by the interface (which uses the API as well), which uses extra queries for pagination and such. I should go grab me a cup of coffee...

Anyways, fixed. Thanks for your time and the sharp eye @felixfontein!